### ISSUE TITLE
regression in main, qt6: library track colors no longer appear

### ISSUE THREAD

**Issue by ywwg** [mixxxdj/mixxx#12355]
### Bug Description

Track colors have disappeared from the track listing in the main branch. I did a git bisect and the commit that broke it was the QT6 checkbox fix: 1ef4e4d6216fafab3b2393cf96942ec23559df60

@fwcd can you take a look?

### Version

HEAD

### OS

Ubuntu 22.04

**Comment by ywwg** [comment]
broken: 
![image](https://github.com/mixxxdj/mixxx/assets/888940/98c12600-3bd3-48ff-a452-9ae2dc925ae4)
working:
![image](https://github.com/mixxxdj/mixxx/assets/888940/796d08e0-9da2-4e72-b308-13aa80f82e43)

**Comment by fwcd** [comment]
Huh, that's very weird. The way Qt applies styles remains a mystery to me.

@ronso0 Any ideas?

**Comment by ronso0** [comment]
So, border 0px for `#LibraryBPMButton::item` is fixing the BPM button, and `WTrackTableView::item` is actually just for the Played checkbox. Disabling the latter should restore the track color bg, except for the BPM column. Is that the case?

**Comment by ronso0** [comment]
Guess we also should take a look at the delegates' paint code. Maybe there's room for optimization.

Oh the Played column doesn't have a delegate.

**Comment by ywwg** [comment]
Setting only 

```#LibraryBPMButton::item {
  border: 0px;
}```

hilariously fixes it for the rest of the row but not the bpm column:

![image](https://github.com/mixxxdj/mixxx/assets/888940/64172171-b6f2-4051-a93e-e10b995999d3)

**Comment by ronso0** [comment]
yes, because `WTrackTableView::item` is the only way to select the Played column but addresses ALL items except the delegates.

I tried a few qss mods last night, but none would fix the icon while having the bg color.
I guess we need to look for the related Qt bug.

**Comment by fwcd** [comment]
Does the highlight color issue occur in Qt 5 builds of `main` too?
